### PR TITLE
test(lib-injection): test telemetry

### PR DESCRIPTION
**Pull request by emmettbutler**
This change adds automated testing of the instrumentation telemetry emitted by the library injection workflow. It takes the approach of setting `DD_TELEMETRY_FORWARDER_PATH` to point to a simple bash script that writes to a file, then checking the contents of that file against per-test expectations about specific telemetry events.

I am planning on adding the following data to telemetry events emitted by ddtrace itself in a separate pull request:
- whether autoinjection was used
- whether the force envvar was set
- whether autoinjection actually installed anything

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

### CODE REVIEW

File: .github/workflows/lib-injection.yml [hunk 1]
```diff
@@ -115,16 +115,16 @@ jobs:
         run: |
           curl http://localhost:18080
           sleep 1  # wait for traces to be sent
+      - name: Check mocked telemetry
```
Comment (brettlangdon) [inline .github/workflows/lib-injection.yml:118]: Is there anything we can print here as confirmation that something actually happened?

Hard to tell from the existing logs: https://github.com/DataDog/dd-trace-py/actions/runs/9389337013/job/25856592783#step:6:3


dumping the telemetry could be good for human audit, wdyt?

**Review by brettlangdon — APPROVED** [review]
suggestion, otherwise lgtm

File: tests/lib-injection/dd-lib-python-init-test-django-unsupported-package-force/validate_telemetry.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import json
+import sys
+
+
+in_text = sys.stdin.read()
+print(in_text)
```
Comment (ZStriker19) [inline tests/lib-injection/dd-lib-python-init-test-django-unsupported-package-force/validate_telemetry.py:6]:
```suggestion
```

File: tests/lib-injection/dd-lib-python-init-test-django-unsupported-package-force/validate_telemetry.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+parsed = json.loads(in_text.split("\t")[-1])
+print(parsed)
```
Comment (ZStriker19) [inline tests/lib-injection/dd-lib-python-init-test-django-unsupported-package-force/validate_telemetry.py:8]:
```suggestion
```

**Review by ZStriker19 — APPROVED** [review]

File: tests/lib-injection/dd-lib-python-init-test-django-unsupported-package-force/validate_telemetry.py (hunk 3 above)
Comment (emmettbutler) [reply tests/lib-injection/dd-lib-python-init-test-django-unsupported-package-force/validate_telemetry.py:8]: These printed outputs are important for human auditability

File: .github/workflows/lib-injection.yml (hunk 1 above)
Comment (emmettbutler) [reply .github/workflows/lib-injection.yml:118]: The validation script prints debug output https://github.com/DataDog/dd-trace-py/actions/runs/9467691845/job/26082304539?pr=9477

**Pull request merged**